core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -672,17 +672,7 @@ public CommitStats commitStats() {
     }
 
     public IndexingStats indexingStats(String... types) {
-        Engine engine = getEngineOrNull();
-        final boolean throttled;
-        final long throttleTimeInMillis;
-        if (engine == null) {
-            throttled = false;
-            throttleTimeInMillis = 0;
-        } else {
-            throttled = engine.isThrottled();
-            throttleTimeInMillis = engine.getIndexThrottleTimeInMillis();
```

Comment:
OK I understand what happened now. Index throttling seemed to be always disabled so I got rid of it. But it was actually not disabled all the time, it was a bug that `InternalEngine#isThrottled` did not override the default behaviour of `Engine#isThrottled`. I opened a PR: https://github.com/elastic/elasticsearch/pull/21592